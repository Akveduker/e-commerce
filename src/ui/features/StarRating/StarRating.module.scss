.container {
    width: 106px;
    height: 20px;
    position: relative;
}
.rating {
    width: 100%;
    height: 100%;
    background: url("../../../assets/icons/ic-actions-star.svg");
    display: block;
    position: absolute;
    left: 0;
    top: 0;
    &--active {
        left: 0;
        top: 0;
        height: 100%;
        display: block;
        background: url("../../../assets/icons/ic-actions-star-active.svg");
        position: absolute;
        &__yellow {
            background: url("../../../assets/icons/ic-actions-star-yellow.svg");
        }
    }
}
