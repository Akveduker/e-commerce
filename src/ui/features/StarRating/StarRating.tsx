import React, { FC } from 'react';
import s from './StarRating.module.scss'
interface StarRatingProps {
    rate: number;
    yellow?: boolean;
}
const StarRating: FC<StarRatingProps> = ({ rate, yellow }) => {
    const className = yellow ? `${s['rating--active']} ${s['rating--active__yellow']}` : s['rating--active']
    return (
        <div className={s['container']}>
            <span className={s['rating']}></span>
            <span className={className} style={{ width: `${rate * 20}%`, }} data-testid='rate'></span>
        </div>
    );
};

export default StarRating;