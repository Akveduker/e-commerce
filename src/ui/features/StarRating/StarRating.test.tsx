import { render, screen } from "@testing-library/react"
import StarRating from "./StarRating"

describe('StarRating', () => {
    test('Снапшот', () => {
        const { container } = render(<StarRating rate={20} />)
        expect(container).toMatchSnapshot()
    })
    test('Класс меняется', () => {
        render(<StarRating rate={30} yellow />)
        expect(screen.getByTestId('rate')).toHaveClass('rating--active__yellow')
    })
})