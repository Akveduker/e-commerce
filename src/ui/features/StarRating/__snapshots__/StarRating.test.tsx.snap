// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRating Снапшот 1`] = `
<div>
  <div
    class="container"
  >
    <span
      class="rating"
    />
    <span
      class="rating--active"
      data-testid="rate"
      style="width: 400%;"
    />
  </div>
</div>
`;
